tests/drivers/adc/adc_api/src/test_adc.c

```diff
@@ -269,7 +277,9 @@ static const struct device *init_adc(void)
 		"Setting up of the second channel failed with code %d", ret);
 #endif /* defined(ADC_2ND_CHANNEL_ID) */
 
-	(void)memset(m_sample_buffer, 0, sizeof(m_sample_buffer));
+	for (i = 0; i < BUFFER_SIZE; ++i) {
```

Comment:
curious, why memset could not work here?

Reply:
Because it writes bytes and the buffer items are of type `int16_t`.

Reply:
wouldn't sizeof operator take care of this?

Reply:
It's not about the size but the value written. `memset()` was okay for writing all zeros, but you cannot write non-zero 16-bit values with it.

Reply:
Right, now I get it. You are actually checking the SHORT MIN values at run time; i thought int was only needed to set the m_sample_buffer contents  initially to non-zero